src/edu/csus/ecs/pc2/imports/ccs/ContestSnakeYAMLLoader.java

```diff
@@ -261,6 +262,20 @@ private void setTitle(IInternalContest contest, String title) {
 
     }
 
+
+    private void setSandboxCommand(IInternalContest contest, String sandboxCommandLine) {
+        ContestInformation contestInformation = contest.getContestInformation();
+        contestInformation.setSandboxCommandLine(sandboxCommandLine);
+        contest.updateContestInformation(contestInformation);
+    }
+    
+    private void setMemoryLimit(IInternalContest contest, int memoryLimit) {
```

Comment:
For consistency with other class setters, as well as to provide clarity for future developers, I think this method should be renamed `setMemoryLimitMB()` (and the _int_ parameter should be similarly renamed `memoryLimitMB`).

Reply:
Changed.  Funny thing is: it's never used anyway... Changed in [this commit](https://github.com/pc2ccs/pc2v9/pull/732/commits/ac45863596ea4b46d29a95e70db66998c8b423d5)

Reply:
> Changed. Funny thing is: it's never used anyway... Changed in [this commit](https://github.com/pc2ccs/pc2v9/pull/732/commits/ac45863596ea4b46d29a95e70db66998c8b423d5)

I find it not just funny, but a bit disturbing that it's "never used".  I am under the impression that YAML support for setting memory limits on problems is supported in this PR.  If that's true (is it?)  then how is it being done without calling this method?

Reply:
The memory limit is set on a per-problem basis by calling the `problem.setMemoryLimitMB(numMeg)`.  There is (currently no way to set the system global memory limit, which is what the `setMemoryLimitMB()` method in `ContestSnakeYamlLoader.java` does.  Presumably, that was the (future) intention of creating that method?

Is that out of scope with this PR?  Should I add a property in system.pc2.yaml file to set it, or, create an issue describing the need for it?

Reply:
I don't think it's out of scope for this PR.  On the contrary, the purpose of [the original issue, 295](https://github.com/pc2ccs/pc2v9/issues/295) was not actually "provide sandbox support", it was "add support for per-problem memory limits (and MLE judgement)".   Ideally,  a PR to do that should cover all aspects of "setting memory limits", and it seems to me that it would be ideal if it followed the same organization as "setting time limits" -- meaning including the ability in YAML to both set a global memory limit and the ability to override that global limit on a per-problem basis, just as can currently be done with time limits.

Having said that, if you don't feel like adding that, I would accept the filing of a new issue.  But it seems to me like the proper time to fix it is "now"?

Reply:
Well, I added the setting of the global memory limit from the yaml in[ this commit] (https://github.com/pc2ccs/pc2v9/pull/732/commits/2b8e65c9a8f6f2e31dc3d0aa0ebbc7eb05c76260) (and tested to make sure it sets the global value) .  HOWEVER: no one actually USES the memory limit in the `ContestInformation `class, it seems.  By default, it is 0 (which is supposed to mean unlimited).  But, no one uses the value.  The only value used is the one from the `problem.yaml `files for each problem.  

It may be time to create another issue to decide exactly how and when the global limit should be used (and, set/viewed from the GUI).  Presumably, it should be used as the "default" value when reading in the `problem.yaml` if a limit is not specified for the problem.  Currently, it just uses zero. (0).  In addition, if a problem is added from the GUI, it should use the global default memory limit if one is not set by the user.  I am sure there are other things that have to be considered as to when to make use of the global memory limit.  At least now it is at least SET from the yaml (which is the only place you CAN set it now).